timeutil/schedule.go

```diff
@@ -708,3 +708,47 @@ func parseEventSet(s string) (*Schedule, error) {
 
 	return &schedule, nil
 }
+
+// Includes checks whether given time t falls inside the time range covered by
+// the schedule. A single time schedule eg. '10:00' is treated as spanning the
+// time <10:00, 10:01)
+func (sched *Schedule) Includes(t time.Time) bool {
+	if len(sched.WeekSpans) > 0 {
+		var weekMatch bool
+		for _, week := range sched.WeekSpans {
+			if week.Match(t) {
+				weekMatch = true
+				break
+			}
+		}
+		if !weekMatch {
+			return false
+		}
+	}
+
+	for _, tspan := range sched.flattenedClockSpans() {
+		window := tspan.Window(t)
+		if window.End.Equal(window.Start) {
+			// schedule granularity is a minute, a schedule '10:00'
+			// in fact is: <10:00, 10:01)
+			window.End = window.End.Add(time.Minute)
+		}
+		// Includes() does the <start,end> check, but we really what
+		// <start,end)
+		if window.Includes(t) && t.Before(window.End) {
```

Comment:
This looks a bit weird, can't window.Includes(..) check the exact range in the first place?

Reply:
> This looks a bit weird, can't window.Includes(..) check the exact range in the first place?

I don't think it looks too weird, but indeed we might change the behavior of Includes to be exclusive on the right. It seems to match the vague feeling of 10:00-11:00, in that people don't really mean 11:00:59 with that.

Reply:
To be clear, this is not a big deal either way. The change might not be worth pursuing right now.

Reply:
I did the change, but I'm leaving it in a separate branch and can do a follow-up later. First, I'd like to land this and the rest of timer services code. 

If you wish to look at the commit, it's right here https://github.com/bboozzoo/snapd/commit/cda6b19b778fade915b49b44e78fc59974c2a15f